lib/loss.c

```diff
@@ -78,7 +78,7 @@ quicly_error_t quicly_loss_detect_loss(quicly_loss_t *loss, int64_t now, uint32_
     /* Mark packets as lost if they are smaller than the largest_acked and outside either time-threshold or packet-threshold
      * windows. Once marked as lost, cc_bytes_in_flight becomes zero. */
     while ((sent = quicly_sentmap_get(&iter))->packet_number != UINT64_MAX) {
-        int64_t largest_acked_signed = loss->largest_acked_packet_plus1[sent->ack_epoch] - 1;
+        int64_t largest_acked_signed = loss->largest_acked_packet_plus1.per_epoch[sent->ack_epoch] - 1;
```

Comment:
It would be a simplification if we could get rid of the per-epoch state, this is the only place that uses it.

The fear that kept us from using `all_` here is that that could lead to the PN-based loss detection calling out reorders as losses. That could happen as one datagram might carry up to 3 QUIC packets, with each packet using a different packet number.

We do not need to come up with a solution now, however note that ideally we should call out losses per datagram (rather than per packet). Using QUIC packets (as described in RFC 9002) is nothing more than approximation.